streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -59,4 +78,78 @@ public SourceNode source(String topic) {
         return stateStoreSuppliers;
     }
 
+
+    /**
+     * Produces a string representation contain useful information about a ProcessorTopology.
+     * This is useful in debugging scenarios. Prints graph by performing a breadth-first search starting
+     * from source nodes
+     * @return A string representation of the ProcessorTopology instance.
+     */
+    public String toString() {
+        Map<String, Boolean> visitedNode = new HashMap<String, Boolean>();
+        Queue queue = new LinkedList();
+        String processorTopologyString = "ProcessorTopology[";
+        Set<SourceNode> tmpSources = sources();
+
+        // pre-process source nodes to get reverse mapping
+        Map<String, String> sourceNameToTopic = new HashMap<>();
+        for (String topic : sourceByTopics.keySet()) {
+            SourceNode source = sourceByTopics.get(topic);
+            sourceNameToTopic.put(source.name(), topic);
+        }
+
+        // pre-process sink nodes to get reverse mapping
+        Map<String, String> sinkNameToTopic = new HashMap<>();
+        for (String topic : sinkByTopics.keySet()) {
+            SinkNode sink = sinkByTopics.get(topic);
+            sinkNameToTopic.put(sink.name(), topic);
+        }
+
+        // add all sources to a queue for Breadth-First traversal
+        if (tmpSources != null) {
+            // add all sources
+            for (String topic : sourceByTopics.keySet()) {
+                SourceNode source = sourceByTopics.get(topic);
+                // mark as visited
+                visitedNode.put(source.name(), true);
+                // put in queue
+                queue.add(source);
+            }
+
+            // iterate to next level
```

Comment:
I tried to print the topology with the current code in WordCountDemo, and here is the output:

```
ProcessorTopology[sourceNode=ProcessorNode[name=KSTREAM-SOURCE-0000000006,sourceTopic=X-KSTREAM-MAP-0000000002-repartition-->node=ProcessorNode[name=KSTREAM-AGGREGATE-0000000003,StateStores[Counts,],-->		
sourceNode=ProcessorNode[name=KSTREAM-SOURCE-0000000000,sourceTopic=streams-file-input-->node=ProcessorNode[name=KSTREAM-FLATMAPVALUES-0000000001,-->		
node=ProcessorNode[name=KTABLE-TOSTREAM-0000000007,-->		
node=ProcessorNode[name=KSTREAM-MAP-0000000002,-->		
node=ProcessorNode[name=KSTREAM-SINK-0000000008,sinkTopic=streams-wordcount-output,-->		
node=ProcessorNode[name=KSTREAM-FILTER-0000000005,-->		
		
node=ProcessorNode[name=KSTREAM-SINK-0000000004,sinkTopic=X-KSTREAM-MAP-0000000002-repartition,-->		
		
]
```